hibernate-core/src/main/java/org/hibernate/tuple/entity/AbstractEntityTuplizer.java

```diff
@@ -404,6 +403,13 @@ public void setIdentifier(Object entity, Serializable id, EntityMode entityMode,
 					if ( association == null ) {
 						// otherwise look for an initialized version
 						association = persistenceContext.getEntity( entityKey );
+						if ( association == null ) {
+							// get the association out of the entity itself
+							association = sessionFactory.getMetamodel().entityPersister( entityName ).getPropertyValue(
+									entity,
+									virtualIdComponent.getPropertyNames()[i]
+							);
+						}
 					}
```

Comment:
Without the code added above, ```org.hibernate.test.annotations.derivedidentities.e1.a.DerivedIdentitySimpleParentIdClassDepTest``` fails because ```association == null```.